styles/responsive_styles.h

```diff
@@ -0,0 +1,68 @@
+
+// These Styles all use Layers<> over a BASE Color/Style
+
+// ResponsiveLockupL<LOCKUP COLOR, TRANSITION1, TRANSITION2, TOP, BOTTOM, SIZE>
+// Implements LocalizedLockup that will move based on the angle of the blade
+// TRANSITION1 & TRANSITION2 = transition Begin & End
+// TOP = uppermost lockup position limit, BOTTOM = lowermost lockup position limit, 32768 = tip, 0 = hilt
+// SIZE controls LOCKUP area size 0 ~ 32768
+
+template<class COLOR, class TR1, class TR2, int TOP = 26000, int BOTTOM = 4000, int SIZE = 10000>
+using ResponsiveLockupL = LockupTrL<AlphaL<COLOR, Bump<Scale<BladeAngle<>,Int<TOP>,Int<BOTTOM>>,Int<SIZE>>>, TR1, TR2, SaberBase::LOCKUP_NORMAL> ; 
+
+// ResponsiveDragL<DRAG COLOR, TRANSTION1, TRANSITION2, SIZE1, SIZE2>
+// Implements Drag that will increase or decrease in size based on turning hilt
+// TRANSITION1 & TRANSITION2 = transition Begin & End
+// SIZE1 & SIZE2 control limits for DRAG size with TwistAngle
+
+template<class COLOR, class TR1, class TR2, int SIZE1 = 2000, int SIZE2 = 10000>
+using ResponsiveDragL = LockupTrL<AlphaL<COLOR, SmoothStep<Int<32000>,Scale<TwistAngle<>,Int<SIZE1>,Int<SIZE2>>>>,TR1, TR2, SaberBase::LOCKUP_DRAG> ;
+
+// ResponsiveMeltL<MELT COLOR, TRANSITION1, TRANSITION2, SIZE1, SIZE2>
+// Implements Melt effect for cutting through object, size will change to mimic metal heating and intensity will increase or decrease based on turning hilt
+// TRANSITION1 & TRANSITION2 = transition Begin & End
+// SIZE1 & SIZE2 control MELT area limits for TwistAngle
+
+template<class COLOR = Mix<TwistAngle<>,OrangeRed,Red>, class TR1 = TrWipeIn<600>, class TR2 = TrWipe<600>, int SIZE1 = 4000, int SIZE2 = 10000 >
+using ResponsiveMeltL = LockupTrL<AlphaL<COLOR, SmoothStep<Int<30000>,Scale<TwistAngle<>,Int<SIZE1>,Int<SIZE2>>>>, TR1, TR2, SaberBase::LOCKUP_MELT> ;
+
+// ResponsiveLightningBlockL<LIGHTNING BLOCK COLOR, TRANSITION1, TRANSITION2>
+// Implements hybrid Force Lightning Block with animation, intensity responds to turning the hilt and location/focus will respond to blade angle
+// TRANSITION1 & TRANSITION2 = transition Begin & End
+
+template<class COLOR, class TR1, class TR2>
+using ResponsiveLightningBlockL = LockupTrL<AlphaL<COLOR, LayerFunctions<Bump<Sin<Int<24>,Scale<BladeAngle<>,Int<14000>,Int<6000>>,Int<4000>>,Sin<Int<17>,Scale<TwistAngle<>,Int<12000>,Int<9000>>,Int<8000>>>,Bump<Sin<Int<27>,Scale<BladeAngle<>,Int<24000>,Int<10000>>,Int<8000>>,Sin<Int<20>,Scale<TwistAngle<>,Int<10000>,Int<7000>>,Int<2000>>>,Bump<Sin<Int<25>,Int<18000>,Scale<BladeAngle<24000,32768>,Int<22000>,Int<15000>>>,Sin<Int<10>,Scale<TwistAngle<>,Int<6000>,Int<1000>>,Int<0>>>,Bump<Sin<Int<31>,Scale<BladeAngle<22000,32768>,Int<30000>,Scale<BladeAngle<24000,32768>,Int<16000>,Int<10000>>>,Scale<BladeAngle<24000,32768>,Int<16000>,Int<8000>>>,Sin<Int<18>,Scale<TwistAngle<>,Int<8000>,Int<2000>>,Int<0>>>>>, TR1, TR2, SaberBase::LOCKUP_LIGHTNING_BLOCK> ;

```

Comment:
This is really hard to read, and thus also hard to update or maintain.
We might want to split it into smaller pieces using aliases, or perhaps just indent it or something.

Reply:
Makes sense, also defaulted TR1 & TR2 to TrInstant